AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/internal/SentryCrashLogging.kt

```diff
@@ -23,28 +26,29 @@ internal class SentryCrashLogging constructor(
                 dsn = dataProvider.sentryDSN
                 environment = dataProvider.buildType
                 release = dataProvider.releaseName
-                setDebug(dataProvider.enableCrashLoggingLogs)
+                isDebug = dataProvider.enableCrashLoggingLogs
                 setTag("locale", dataProvider.locale?.language ?: "unknown")
                 beforeSend = SentryOptions.BeforeSendCallback { event, _ ->
 
                     if (!dataProvider.crashLoggingEnabled()) return@BeforeSendCallback null
 
                     dropExceptionIfRequired(event)
                     appendExtra(event)
-                    appendApplicationContext(event)
-                    appendUser(event)
                     event
                 }
             }
         }
-    }
-
-    private fun appendUser(event: SentryEvent) {
-        event.user = dataProvider.userProvider()?.toSentryUser()
-    }
 
-    private fun appendApplicationContext(event: SentryEvent) {
-        event.appendTags(dataProvider.applicationContextProvider())
+        applicationScope.launch {
+            dataProvider.user.collect {
+                sentryWrapper.setUser(it?.toSentryUser())
+            }
+        }
+        applicationScope.launch {
+            dataProvider.applicationContextProvider.collect {
+                sentryWrapper.setTags(it)
+            }
+        }
```

Comment:
Hotspot of the PR: here, we observe incoming new data and update Sentry, independently from `SentryOptions.BeforeSendCallback` callback.